frontend/js/src/search/SongSearch.tsx

```diff
@@ -0,0 +1,166 @@
+    "artist-credit": {
+      name: string;
+      joinphrase?: string;
+      artist: {
+        id: string;
+        name: string;
+      };
+    }[];
+    releases: {
+      id: string;
+      title: string;
+      "artist-credit": {
+        name: string;
+        joinphrase?: string;
+      }[];
+      "release-groups": {
+        id: string;
+        title: string;
+      }[];
+    }[];
+  }[];
+};
+
+export default function SongSearch(props: SongSearchProps) {
+  const { APIService } = React.useContext(GlobalAppContext);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const searchParamsObj = getObjectForURLSearchParams(searchParams);
+  const currPageNoStr = searchParams.get("page") || "1";
+  const currPageNo = parseInt(currPageNoStr, 10);
+
+  const { searchQuery } = props;
+
+  const { data: loaderData, isLoading: loading } = useQuery({
+    queryKey: ["search-song", searchQuery, currPageNoStr],
+    queryFn: async () => {
+      try {
+        const offset = (currPageNo - 1) * RECORDING_COUNT_PER_PAGE;
+        const queryData = await APIService.recordingLookup(
+          searchQuery,
+          offset,
+          RECORDING_COUNT_PER_PAGE
+        );
+        return {
+          data: queryData as SongTypeSearchResult,
+          hasError: false,
+          errorMessage: "",
+        };
+      } catch (error) {
+        return {
+          data: {} as SongTypeSearchResult,
+          hasError: true,
+          errorMessage: error.message,
+        };
+      }
+    },
+  });
+
+  const {
+    data: rawData = {} as SongTypeSearchResult,
+    hasError = false,
+    errorMessage = "",
+  } = loaderData || {};
+
+  const { recordings = [] } = rawData;
+  const totalPageCount = Math.ceil(rawData.count / RECORDING_COUNT_PER_PAGE);
+
+  const handleClickPrevious = () => {
+    setSearchParams({
+      ...searchParamsObj,
+      page: Math.max(currPageNo - 1, 1).toString(),
+    });
+  };
+
+  const handleClickNext = () => {
+    setSearchParams({
+      ...searchParamsObj,
+      page: Math.min(currPageNo + 1, totalPageCount).toString(),
+    });
+  };
+
+  const getListenCard = (recording: SongTypeSearchResult["recordings"][0]) => {
+    const artists: MBIDMappingArtist[] = [];
+    const artistMBIDs: Array<string> = [];
+    const artistCredit = recording["artist-credit"];
+    artistCredit.map((ac) => {
+      artists.push({
+        artist_mbid: ac.artist.id,
+        artist_credit_name: ac.name,
+        join_phrase: ac.joinphrase ?? "",
+      });
+      artistMBIDs.push(ac.artist.id);
+      return null;
+    });
+
+    return (
+      <ListenCard
+        key={recording.id}
+        listen={{
+          listened_at: 0,
+          track_metadata: {
+            mbid_mapping: {
+              artists,
+              artist_mbids: artistMBIDs,
+              release_mbid:
+                recording.releases?.length > 0 ? recording.releases[0].id : "",
+              recording_mbid: recording.id,
+            },
```

Comment:
I don't know for sure that we want to supply mbid_mapping here, considering it is usually provided by the server rather than manually added.
Semantically, this data we have is not from our MBID mapping, so i would avoid it.

Reply:
In case a track has multiple artists, We would need to supply mbid_mapping here. 

Without mbid_mapping: 
<img width="713" alt="image" src="https://github.com/metabrainz/listenbrainz-server/assets/84081628/0f7e6380-e0ab-4a15-b3a3-7fbf934df643">
With mbid_mapping:
<img width="738" alt="image" src="https://github.com/metabrainz/listenbrainz-server/assets/84081628/4a047bf7-9a5b-44c8-a315-b95168df7ed5">

Reply:
I see, thanks for the clarification.
Indeed that feature relies entirely on the mbid_mapping data, which is not ideal but also probably not something we should try to change in this PR,
Objection withdrawn! :D